### PR TITLE
Easy suit sensor reminder alert prompt

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Players will now have an easy alert they can click on to adjust their suit sensors when either the AI over vox or someone with a megaphone mentions them, anecdotally the two inevitabilities. ~~The first time a user adjusts their sensors (doesn't matter how), they will no longer get the prompt.~~ Alert only shows the very first time.

## Why It's Good For The Game
Adjusting suit sensors is a weird process; opening inventory, right clicking on your jumpsuit, and clicking Adjust Suit Sensors, or typing in the verb in the command bar. This is exemplified for new players, as having to right click on something in the first place isn't that common, and these messages about suit sensors are as meaningful as the ambience telling players to go to the disembarkation lounge. This just adds basic quality of life that helps both new and old players while also ideally having little overhead and annoyance.

## Changelog
:cl:
add: Added an alert prompt to easily adjust suit sensors when reminded.
/:cl:

### CODE REVIEW

**Comment by HomocidlePancakes** [comment]
Can't tell if it's super good or super dumb

**Comment by terranaut1** [comment]
at that point just make it a game preference to have your suit sensors start at X level. people forgetting about them is part of what lets people get away with murder.

**Comment by 4dplanner** [comment]
jared-fogle wants to be able to track you more easily

**Comment by kriskog** [comment]
I'm all for UI improvements and making suit sensors easier to find for new players, but this is a step too far. Do note you can find the suit sensors verb in the objects tab.

Also, does this notification take focus and hold my inputs hostage?

**Comment by Jared-Fogle** [comment]
> Also, does this notification take focus and hold my inputs hostage?
Shouldn't? You have to click on it, it's the exact same as the alerts for something like object of interests.

**Comment by kriskog** [comment]
Yeah, you added a video. Only a pop up would do that.

**Comment by ATH1909** [comment]
Make the alert only pop up once ever per round (for any given person) period. In other words, if they miss the alert (and didn't adjust their sensors), they still shouldn't ever get it again.

**Comment by Jared-Fogle** [comment]
> Make the alert only pop up once ever per round (for any given person) period. In other words, if they miss the alert (and didn't adjust their sensors), they still shouldn't ever get it again.
Works for me.

File: code/__DEFINES/components.dm [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #define COMSIG_GLOB_MOB_CREATED "!mob_created"					//mob was created somewhere : (mob)
 #define COMSIG_GLOB_MOB_DEATH "!mob_death"						//mob died somewhere : (mob , gibbed)
 #define COMSIG_GLOB_LIVING_SAY_SPECIAL "!say_special"			//global living say plug - use sparingly: (mob/speaker , message)
+#define COMSIG_GLOB_AI_VOX "!ai_vox"							//ai used vox announcements : (mob/living/silicon/ai, list/words)
```
Comment (chesse20) [inline code/__DEFINES/components.dm]: looks good

File: code/datums/components/suit_sensors_reminder.dm [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	if(!istype(U))
+		return
+	last_used = world.time
+	var/obj/screen/alert/toggle_suit_sensors/notification = H.throw_alert("toggle_suit_sensors", /obj/screen/alert/toggle_suit_sensors)
+	if(!notification)
+		return
+	var/ui_style = H.client?.prefs?.UI_style
+	if(ui_style)
+		notification.icon = ui_style2icon(ui_style)
+	notification.desc = "[source] wants to remind you to toggle your suit sensors so that the medical staff can find you in the case of an emergency."
```
Comment (chesse20) [inline code/datums/components/suit_sensors_reminder.dm]: good desc

**Review by chesse20 — APPROVED** [review]
Looks good to me lets merge it

**Comment by ExcessiveUseOfCobblestone** [comment]
@chesse20 stop

**Comment by MrPerson** [comment]
The set suit sensors verb sucks and should be removed outright instead

**Comment by Barhandar** [comment]
>Also, I'm still not sure how I feel about this. It makes not turning up your suit sensors to max something that's a lot harder for you to "forget" to do (and thus more suspicious). Suit sensors are set to random tracking levels at roundstart for a reason, y'know.

Agreed, this is rather hugboxy AND can interfere with antags.

**Comment by Jared-Fogle** [comment]
"This makes it harder to lie as antag" is a much better argument than "I don't want to fix bad UX because then a new player might figure it out", but it's not hard to ignore this. It shows once now, so if it's said round start (like it usually is), and it only shows for a short timespan, then you might be setting up your department, see it, think "eh I'll do it later, setting up [SM/medbay/etc] is more important", and then forget because either more things come up or you're just forgetful.

**Comment by ExcessiveUseOfCobblestone** [comment]
tfw there's a feature that allows people to track both your health and location in real time with the only counter being an antag kills them fast enough that the damage breaks the sensors before someone is able to utilize the information.

Literally the only balancing act is you have to monitor it and the client has to max it out.

**Comment by Akrilla** [comment]
I don't like the idea of the UI element only appearing once someone else says "suit sensors". Just add the element to the bar at the top left, like with the chameleon jumpsuit etc.

**Comment by Dorsisdwarf** [comment]
@Akrilla the dynamic action button list is already crowded most of the time, adding a button which will be on it 24/7 wont really improve that

**Comment by SpaceManiac** [comment]
This is a really funny idea but

1. An alert being thrown by overhearing a specific word or phrase is a totally new kind of UI and should be justified more strongly (does this make sense to do for anything else?)
2. People forgetting or failing to turn on suit sensors is part of what keeps their power in check

**Comment by Jared-Fogle** [comment]
@SpaceManiac 

> An alert being thrown by overhearing a specific word or phrase is a totally new kind of UI and should be justified more strongly (does this make sense to do for anything else?)

Maybe, I can't think of anything else it'd be useful for right now--perhaps job specific notifications? I don't believe this is very intrusive, especially after the change to make it only show once.

> People forgetting or failing to turn on suit sensors is part of what keeps their power in check

I'm not totally convinced this makes it that much harder to forget/fail to do, this applies to AI vox messages that everyone can hear and megaphone messages whose whole purpose is to stand out in the chat box. It doesn't show for that much longer than I'd expect someone's attention span to, and can be further lowered. This also forfeits a point that's out of scope for this PR that suit sensors just seem badly designed if that's a large factor to keeping them in check (see @ExcessiveUseOfCobblestone's comment).

**Comment by fludd12** [comment]
You could make it so it only works for megaphones, and have it make `suit sensors` a link that opens the suit sensors popup. It would also mean you wont have to make a list of people who've already set them.

**Comment by Jared-Fogle** [comment]
> You could make it so it only works for megaphones, and have it make suit sensors a link that opens the suit sensors popup. 

I considered this but I'm not sure it's much better, and also I think it's largely more difficult.

> It would also mean you wont have to make a list of people who've already set them.

It doesn't do this now anyway.

**Comment by anconfuzedrock** [comment]
when do we ban chesse from the repository he's a "contributor" because he made a pr adding a literal furry orderable from cargo that had to be reverted and all of his other prs are intended to rile people up with shit like damn daniel shoes and the ability to pee your pants

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm not interested in suit sensors so find someone (else) to merge this soon.

**Comment by Jared-Fogle** [comment]
Not sure if that means I'm meant to just sit on this or I have to actively look for someone to merge it?

**Comment by ExcessiveUseOfCobblestone** [comment]
I would check to see if someone is interested in this else it's getting closed

**Comment by Jared-Fogle** [comment]
Doesn't sound worth the effort.

**Pull request closed without merging**